utils/preferences/tests/dtf.rs

```diff
@@ -0,0 +1,85 @@
+use icu_datetime::options::preferences::HourCycle;
+use icu_locid::{unicode_ext_key, Locale};
+use icu_preferences::preferences;
+
+#[derive(Clone, Copy)]
+enum Calendar {
+    Gregory,
+}
+
+impl TryFrom<&icu_locid::extensions::unicode::Value> for Calendar {
+    type Error = ();
+
+    fn try_from(_i: &icu_locid::extensions::unicode::Value) -> Result<Self, Self::Error> {
+        Ok(Calendar::Gregory)
+    }
+}
+
+preferences!(
+    DTFPreferences,
+    ResolvedDTFPreferences,
+    {
+        hour_cycle => Option<HourCycle>, HourCycle, Some(unicode_ext_key!("hc")),
+        calendar => Option<Calendar>, Calendar, Some(unicode_ext_key!("ca"))
+    }
+);
+
+struct DateTimeFormat {
+    prefs: ResolvedDTFPreferences,
+}
+
+impl DateTimeFormat {
+    pub fn new(locale: &Locale, prefs: &DTFPreferences) -> Self {
```

Comment:
Thought: I want to make the `Locale` argument be a trait (see #1662). Should we consider making that trait also encompass the preferences bag? For example, plain `Locale` can implement the trait, and so can `DTFPreferences`, which would wrap the `Locale` and add extra overrides.

Reply:
I think that preferences shouldn't be folded into `Locale`. Preferences can be resolved without locale, and I am planning to make the resolver/constructor also accept `None` for `Locale` and just resolve between Preferences and ResolvedPreferences.